### PR TITLE
Optimise swoole-pg

### PR DESCRIPTION
**Pull request by doubaokun**
Optimise `Swoole-postgres`.

### CODE REVIEW

File: frameworks/PHP/swoole/swoole-server.php [hunk 1]
```diff
@@ -38,15 +38,15 @@
     // Create an array with the response string.
     $arr = [];
 
-    $db->prepare('s', 'SELECT id, randomnumber FROM World WHERE id = $1');
+    $db->s = $db->s ?? $db->prepare('s', 'SELECT id, randomnumber FROM World WHERE id = $1');
```
Comment (jcheron) [inline frameworks/PHP/swoole/swoole-server.php:41]: `$db->s` is useless and could be removed.

This is enouth, no?
```php
$db->prepare('s', 'SELECT id, randomnumber FROM World WHERE id = $1');
```

File: frameworks/PHP/swoole/swoole-server.php [hunk 2]
```diff
@@ -72,7 +72,7 @@
 
     $fortune = [];
 
-    $db->prepare('f', 'SELECT id, message FROM Fortune');
+    $db->f = $db->f ?? $db->prepare('f', 'SELECT id, message FROM Fortune');
```
Comment (jcheron) [inline frameworks/PHP/swoole/swoole-server.php:75]: Same thing:
```php
$db->prepare('f', 'SELECT id, message FROM Fortune');
```

File: frameworks/PHP/swoole/swoole-server.php [hunk 3]
```diff
@@ -113,16 +113,16 @@
 
     $arr = [];
 
-    $db->prepare('us', 'SELECT randomnumber FROM World WHERE id = $1');
-    $db->prepare('uu', 'UPDATE World SET randomnumber = $1 WHERE id = $2');
+    $db->us = $db->us ?? $db->prepare('us', 'SELECT randomnumber FROM World WHERE id = $1');
+    $db->uu = $db->uu ?? $db->prepare('uu', 'UPDATE World SET randomnumber = $1 WHERE id = $2');
```
Comment (jcheron) [inline frameworks/PHP/swoole/swoole-server.php:117]:
```php
$db->prepare('us', 'SELECT randomnumber FROM World WHERE id = $1');
$db->prepare('uu', 'UPDATE World SET randomnumber = $1 WHERE id = $2');
```

**Review by jcheron — CHANGES_REQUESTED** [review]
Either we use the named prepared queries of **Swoole_postgresql**:
```php
$db->prepare('us', 'SELECT randomnumber FROM World WHERE id = $1');
$res = $db->execute('us', [$id]);
```
Either we store in a variable the statement:
```php
$db->s ??= $db->prepare('SELECT randomnumber FROM World WHERE id = $1');
$res = $db->s->execute([$id]);
```
but not both mixed

File: frameworks/PHP/swoole/swoole-server.php (hunk 1 above)
Comment (doubaokun) [reply frameworks/PHP/swoole/swoole-server.php:41]: @jcheron Thanks for the review, but`$db->s` is used to store states, it is initialized only once across multiple requests and added by purpose. 

There is no such API `$db->s->execute([$id]);`.

Same thing for the following suggestions.

Comment (jcheron) [reply frameworks/PHP/swoole/swoole-server.php:41]: I can' t see where are used `$db->us`, `$db->uu` and `$db->s`...
The initialization of `$db->s` and others seems to be unused afterwards.

we can assume that this code below already does this initialization work for each worker:
```php
$db->prepare('s', 'SELECT id, randomnumber FROM World WHERE id = $1');
```

And that it is possible to run the prepared query `s` several times this way:
```php
$res = $db->execute('s', [$id]);
```

Comment (doubaokun) [reply frameworks/PHP/swoole/swoole-server.php:41]: Hi @jchannon, these variable is for checking if the `prepare` query is executed for once. For performance reason, these `prepare` query are only executed once for all the requests after the change. Hope this make sense?

Comment (jcheron) [reply frameworks/PHP/swoole/swoole-server.php:41]: Hi @doubaokun 
I modified my previous comment after testing

**Comment by doubaokun** [comment]
@jcheron updated to the shorter form.

**Review by jcheron — APPROVED** [review]

**Pull request merged**